app/api/delete-event/route.ts

```diff
@@ -0,0 +1,50 @@
+import { getAppServerSession, isAdmin } from "@/app/auth"; // ensuring that only trusted users can access
+import { NextResponse } from "next/server"; // structuring response to users
+import { prisma } from "../db"; // bridge between me and db
+
+/*
+    1. authenticate user: allowDelete
+    2. delete the event from db: deleteEvent
+    3. reflect change to user interface: DELETE
+*/
+
+async function allowDelete(eventId: number, userEmail: string): Promise<boolean> {
+    if (isAdmin(userEmail)) return true;
+    const event = await prisma.event.findUnique({
+        where: { id: eventId },
+        select: { fromEmail: { select: { senderEmail: true } } }
+    });
+    if (!event) return false;
+    return event.fromEmail?.senderEmail === userEmail;
+}
+
+async function deleteEvent(eventId: number): Promise<void> {
+    try {
+        await prisma.event.delete({
+            where: { id: eventId }
+        });
+        console.log(`Successfully deleted event ${eventId}.`)
+    } catch (error) {
+        console.error(`Error deleting the event ${eventId}: `, error); // tells the user what the error is
+        throw new Error(`Failed to delete event ${eventId}.`); // throws an error to stop the execution
```

Comment:
This is a minor detail here, but while this `throw` statement will indeed stop the execution of the `deleteEvent` function, but it's important to note that it doesn't automatically stop the execution of the entire API route. The error will be propagated to the caller (the DELETE function), but it's not being caught or handled there.

Consider adding a try/catch block in the DELETE function to handle this error gracefully. 

```js
export async function DELETE(request: Request) {
// ... existing code ...
  try {
    await deleteEvent(id);
    return NextResponse.json({ success: true });
    } catch (error) {
    console.error(Error in DELETE route: ${error});
    return NextResponse.json("Internal server error", { status: 500 });
    }
}
```

This way, if an error occurs during the deletion process, we can return an appropriate error response to the client instead of potentially crashing the server or leaving the request hanging. This also makes the `response.ok` block down the line make more sense.